### ISSUE TITLE
ValueError: astimezone() cannot be applied to a naive datetime

### ISSUE THREAD

**Issue by vog** [dmedvinsky/mutt-ics#17]
I get the following error with Python 2 as well as Python 3:

```
Traceback (most recent call last):
  File "mutt_ics.py", line 158, in <module>
    entry_point()
  File "mutt_ics.py", line 154, in entry_point
    return main(sys.argv)
  File "mutt_ics.py", line 148, in main
    output = get_interesting_stuff(cal)
  File "mutt_ics.py", line 56, in get_interesting_stuff
    c = get_component(component)
  File "mutt_ics.py", line 69, in get_component
    return get_event(component)
  File "mutt_ics.py", line 132, in get_event
    result = filter(bool, [get_header(e),
  File "mutt_ics.py", line 103, in get_header
    get_val('DTSTART', format_date)
  File "mutt_ics.py", line 99, in get_val
    vals.append((name_map[name], f(e[name])))
  File "mutt_ics.py", line 80, in format_date
    date_or_time = x.dt.astimezone(tz.tzlocal())
ValueError: astimezone() cannot be applied to a naive datetime
```

ICS file:

```
BEGIN:VCALENDAR
METHOD:REQUEST
PRODID:Microsoft Exchange Server 2010
VERSION:2.0
BEGIN:VTIMEZONE
TZID:W. Europe Standard Time
BEGIN:STANDARD
DTSTART:16010101T030000
TZOFFSETFROM:+0200
TZOFFSETTO:+0100
RRULE:FREQ=YEARLY;INTERVAL=1;BYDAY=-1SU;BYMONTH=10
END:STANDARD
BEGIN:DAYLIGHT
DTSTART:16010101T020000
TZOFFSETFROM:+0100
TZOFFSETTO:+0200
RRULE:FREQ=YEARLY;INTERVAL=1;BYDAY=-1SU;BYMONTH=3
END:DAYLIGHT
END:VTIMEZONE
BEGIN:VEVENT
DESCRIPTION;LANGUAGE=de-DE:\n
DTSTART;TZID=W. Europe Standard Time:20170317T113000
DTEND;TZID=W. Europe Standard Time:20170317T120000
UID:040000008200E00074C5B7101A82E0080000000070D49594829ED201000000000000000
 0100000004E818B43FD05E041B4860AB201D19D00
CLASS:PUBLIC
PRIORITY:5
DTSTAMP:20170316T172442Z
TRANSP:OPAQUE
STATUS:CONFIRMED
SEQUENCE:0
LOCATION;LANGUAGE=de-DE:Telko
X-MICROSOFT-CDO-APPT-SEQUENCE:0
X-MICROSOFT-CDO-OWNERAPPTID:370161633
X-MICROSOFT-CDO-BUSYSTATUS:TENTATIVE
X-MICROSOFT-CDO-INTENDEDSTATUS:BUSY
X-MICROSOFT-CDO-ALLDAYEVENT:FALSE
X-MICROSOFT-CDO-IMPORTANCE:1
X-MICROSOFT-CDO-INSTTYPE:0
X-MICROSOFT-DISALLOW-COUNTER:FALSE
BEGIN:VALARM
DESCRIPTION:REMINDER
TRIGGER;RELATED=START:-PT15M
ACTION:DISPLAY
END:VALARM
END:VEVENT
END:VCALENDAR
```

**Comment by dmedvinsky** [comment]
That's odd. I can't reproduce the exception on your sample. Can you provide some further details? Maybe the icalendar dependency version?

**Comment by vog** [comment]
I tested with Python 3.5.3 using icalendar 3.8.

**Comment by aschiweck** [comment]
Same Issue here: Python 2.7.13, icalendar 3.11.3:

> BEGIN:VCALENDAR
VERSION:2.0
BEGIN:VEVENT
DTSTAMP:20170515T081600Z
DTSTART:20170516T090000
DTEND:20170516T120000
SUMMARY:Some Summary
URL:https://domain.tld
DESCRIPTION:Some Description
LOCATION:Some location
END:VEVENT
END:VCALENDAR

Looks more like a Problem of format_date to me.

**Comment by chelmuth** [comment]
Recently mutt-ics stopped working for me due to this issue. Any news? Anything I could help with?

**Comment by chelmuth** [comment]
Sry, I missed to add that the ICS above works but the following breaks.

```
BEGIN:VCALENDAR
VERSION:2.0
METHOD:PUBLISH
PRODID:Data::ICal 0.22
BEGIN:VEVENT
CATEGORIES:Reise\, Flug
CLASS:PUBLIC
DESCRIPTION:Test
DTEND:20181211T172500
DTSTAMP:20181102T134716Z
DTSTART:20181211T161500
LOCATION:Von Hier
PRIORITY:5
SEQUENCE:0
SUMMARY:Flug nach Da
TRANSP:OPAQUE
UID:NOPE
END:VEVENT
END:VCALENDAR
```

**Comment by dmedvinsky** [comment]
Welp, still works for me:

```
dmedvinsky at loki in ~/s/u/mutt-ics (master %=) in mutt-ics environment
><((°> cat samples/6.ics
BEGIN:VCALENDAR
VERSION:2.0
METHOD:PUBLISH
PRODID:Data::ICal 0.22
BEGIN:VEVENT
CATEGORIES:Reise\, Flug
CLASS:PUBLIC
DESCRIPTION:Test
DTEND:20181211T172500
DTSTAMP:20181102T134716Z
DTSTART:20181211T161500
LOCATION:Von Hier
PRIORITY:5
SEQUENCE:0
SUMMARY:Flug nach Da
TRANSP:OPAQUE
UID:NOPE
END:VEVENT
END:VCALENDAR

dmedvinsky at loki in ~/s/u/mutt-ics (master %=) in mutt-ics environment
><((°> ./mutt_ics/mutt_ics.py samples/6.ics
Subject:  Flug nach Da
Start:    Tuesday, 11 December 2018, 16:15 MSK
End:      Tuesday, 11 December 2018, 17:25 MSK
Location: Von Hier
Description:

Test
```

There must be some difference in the environment, I guess. Maybe it's different local timezone?

**Comment by chelmuth** [comment]
What I did (and may be wrong) with the data above.
```
> sudo -H pip install mutt-ics
/usr/local/lib/python2.7/dist-packages/pip/_vendor/requests/__init__.py:83: RequestsDependencyWarning: Old version of cryptography ([1, 2, 3]) may cause slowdown.
  warnings.warn(warning, RequestsDependencyWarning)
Collecting mutt-ics
Requirement already satisfied: icalendar>=3.9.0 in /usr/local/lib/python2.7/dist-packages (from mutt-ics) (4.0.3)
Requirement already satisfied: python-dateutil in /usr/local/lib/python2.7/dist-packages (from icalendar>=3.9.0->mutt-ics) (2.7.5)
Requirement already satisfied: pytz in /usr/local/lib/python2.7/dist-packages (from icalendar>=3.9.0->mutt-ics) (2018.7)
Requirement already satisfied: six>=1.5 in /usr/local/lib/python2.7/dist-packages (from python-dateutil->icalendar>=3.9.0->mutt-ics) (1.12.0)
Installing collected packages: mutt-ics
Successfully installed mutt-ics-0.8

> mutt-ics < /tmp/ics 
Traceback (most recent call last):
  File "/usr/local/bin/mutt-ics", line 11, in <module>
    sys.exit(entry_point())
  File "/usr/local/lib/python2.7/dist-packages/mutt_ics/mutt_ics.py", line 154, in entry_point
    return main(sys.argv)
  File "/usr/local/lib/python2.7/dist-packages/mutt_ics/mutt_ics.py", line 148, in main
    output = get_interesting_stuff(cal)
  File "/usr/local/lib/python2.7/dist-packages/mutt_ics/mutt_ics.py", line 56, in get_interesting_stuff
    c = get_component(component)
  File "/usr/local/lib/python2.7/dist-packages/mutt_ics/mutt_ics.py", line 69, in get_component
    return get_event(component)
  File "/usr/local/lib/python2.7/dist-packages/mutt_ics/mutt_ics.py", line 132, in get_event
    result = filter(bool, [get_header(e),
  File "/usr/local/lib/python2.7/dist-packages/mutt_ics/mutt_ics.py", line 103, in get_header
    get_val('DTSTART', format_date)
  File "/usr/local/lib/python2.7/dist-packages/mutt_ics/mutt_ics.py", line 99, in get_val
    vals.append((name_map[name], f(e[name])))
  File "/usr/local/lib/python2.7/dist-packages/mutt_ics/mutt_ics.py", line 80, in format_date
    date_or_time = x.dt.astimezone(tz.tzlocal())
ValueError: astimezone() cannot be applied to a naive datetime

> locale
LANG=en_DK.UTF-8
LANGUAGE=en_US
LC_CTYPE=de_DE.UTF-8
LC_NUMERIC=de_DE.UTF-8
LC_TIME=de_DE.UTF-8
LC_COLLATE=C
LC_MONETARY=de_DE.UTF-8
LC_MESSAGES=en_DK.UTF-8
LC_PAPER=de_DE.UTF-8
LC_NAME=de_DE.UTF-8
LC_ADDRESS=de_DE.UTF-8
LC_TELEPHONE=de_DE.UTF-8
LC_MEASUREMENT=de_DE.UTF-8
LC_IDENTIFICATION=de_DE.UTF-8
LC_ALL=

> lsb_release -a
No LSB modules are available.
Distributor ID:	Ubuntu
Description:	Ubuntu 16.04.5 LTS
Release:	16.04
Codename:	xenial
```

May the PIP be outdated?

**Comment by chelmuth** [comment]
The master version also shows the error.
```
> git clone https://github.com/dmedvinsky/mutt-ics.git mutt-ics.git
> cp /tmp/ics mutt-ics.git/samples/6.ics

> ./mutt-ics.git/mutt_ics/mutt_ics.py < mutt-ics.git/samples/5.ics 
Subject:   test event
Organizer: trhd@iki.fi
Start:     Saturday, 24 December 2016, 00:00 
End:       Monday, 26 December 2016, 00:00 
Location:  
Description:

foobar

> ./mutt-ics.git/mutt_ics/mutt_ics.py < mutt-ics.git/samples/6.ics 
Traceback (most recent call last):
  File "./mutt-ics.git/mutt_ics/mutt_ics.py", line 158, in <module>
    entry_point()
  File "./mutt-ics.git/mutt_ics/mutt_ics.py", line 154, in entry_point
    return main(sys.argv)
  File "./mutt-ics.git/mutt_ics/mutt_ics.py", line 148, in main
    output = get_interesting_stuff(cal)
  File "./mutt-ics.git/mutt_ics/mutt_ics.py", line 56, in get_interesting_stuff
    c = get_component(component)
  File "./mutt-ics.git/mutt_ics/mutt_ics.py", line 69, in get_component
    return get_event(component)
  File "./mutt-ics.git/mutt_ics/mutt_ics.py", line 132, in get_event
    result = filter(bool, [get_header(e),
  File "./mutt-ics.git/mutt_ics/mutt_ics.py", line 103, in get_header
    get_val('DTSTART', format_date)
  File "./mutt-ics.git/mutt_ics/mutt_ics.py", line 99, in get_val
    vals.append((name_map[name], f(e[name])))
  File "./mutt-ics.git/mutt_ics/mutt_ics.py", line 80, in format_date
    date_or_time = x.dt.astimezone(tz.tzlocal())
ValueError: astimezone() cannot be applied to a naive datetime
```

**Comment by dmedvinsky** [comment]
Should be fixed by #18.

**Comment by chelmuth** [comment]
Great, I confirm that works for me now with 0.9 from PIP.
